docs(next): Migration instruction (#1446)
huntabyte/shadcn-svelte: sites/docs/src/content/migration/svelte-5/index.md

---
title: Svelte 5 Migration
description: How to migrate to Svelte 5 from Svelte 4.
---

## Prerequisites

1. Make sure you have read up on the changes from Svelte 4 to Svelte 5. Svelte provides a comprehensive guide for this on their [website](https://svelte.dev/docs/svelte/v5-migration-guide).

> 2. Make sure you have committed any pending changes so that you don't lose any code.

Reviewer: ```suggestion
2. Commit any pending changes so that you don't lose any code.
```

> 3. Understand which of your components have custom behavior/styles so that you can reimplement that yourself.

Reviewer: ```suggestion
3. Determine which of your components have custom behavior/styles so you can reimplement those after updating.
```

<script>
    import { Steps, PMExecute, PMInstall, PMRemove } from "$lib/components/docs";
</script>

## Migrate Configs

The `tailwind.config`, `components.json`, and `utils` files have all changed for Svelte 5.

### Automatic

Note: This works best for projects that have not changed the contents of `tailwind.config`, `utils`, and the global css file.

<PMExecute command="shadcn-svelte@next init"/>

### Manual

<Steps>

### Update `components.json`

> Update the registry and add the `hooks` key under `aliases`.

Reviewer: ```suggestion
Add the `registry` to the root object, and add `hooks` and `ui` keys  under `aliases`.
```
